### PR TITLE
Bug 1061548 add user metadata to question onpost

### PR DESCRIPTION
**Pull request by ossreleasefeed**
@Rik r? just opening for now, will clean up the commit bits once we reach this PR

### CODE REVIEW

File: app/js/sumo_db.js [hunk 1]
```diff
@@ -155,7 +177,36 @@
...
+      var set_metadata = endpoint + 'set_metadata/';
+      delete_metadata += '?format=json'; // TODO bug 1088014
+      set_metadata += '?format=json'; // TODO bug 1088014
+
+      return request_with_auth(delete_metadata, 'POST', metadata)
+        .then(function(response) {
+          return request_with_auth(set_metadata, 'POST', metadata)
+            .then(JSON.parse);
+        }, function(error) {
+          // currently if the metadata item did not exist, the server
```
Comment (Rik) [inline app/js/sumo_db.js]: Can you reference the bug number in the comment?

**Comment by Rik** [comment]
Previous commit with my comments: https://github.com/ossreleasefeed/buddyup/commit/025c5fbea8fa7ff16d1111465adcaeec612bb5af

**Comment by ossreleasefeed** [comment]
Ok, need to run this through couple of sanity checks and tests and then it should be ready for re-review

File: app/js/sumo_db.js [hunk 2]
```diff
@@ -194,6 +219,35 @@
             .then(JSON.parse);
           }
         });
+    },
+    /**
+    * Updates the metadata for a specific question id.
+    * @param {int} question_id - ID of the question to update
+    * @param {object} metadata - The metadata item to update
+    */
+    update_metadata: function(question_id, metadata) {
```
Comment (Rik) [inline app/js/sumo_db.js]: update_user_metadata

Comment (Rik) [inline app/js/sumo_db.js]: Grrr, I meant upadte_question_metadata.

File: app/js/utils.js [hunk 3]
```diff
@@ -51,17 +51,24 @@
...
      */
     get_user_meta: function() {
-      var gecko = navigator.userAgent.match(/Gecko\/(\d+\.?)+\d+/g)[0];
-      var os_version = gecko.substr(gecko.indexOf('/') + 1);
-      var handset_type = 'Alcatel'; // TODO: @see http://mzl.la/1y8b4ho
-      var operator = 'Movistar'; // TODO: @see http://mzl.la/1y8b4ho
+      var gecko = navigator.userAgent.match(/rv:([\d\.]+)/)[1];
+      var os_version = {
+        name: 'os_version',
+        value: gecko.substr(gecko.indexOf('/') + 1)
```
Comment (Rik) [inline app/js/utils.js]: This is not needed, gecko is ready to be used without a substr.

Comment (ossreleasefeed) [inline app/js/utils.js]: yep, updated

**Pull request merged**